speechbrain/lobes/models/discrete/sq_codec.py

```diff
@@ -0,0 +1,1394 @@
+        ---------
+        x : torch.Tensor
+            Input tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            Processed output tensor.
+        """
+        x = torch.transpose(x, 1, 2)
+        B, T, C = x.size()
+        x = x.repeat(1, 1, self.num_samples).view(B, -1, C)
+        x = torch.transpose(x, 1, 2)
+        output = self.activation(self.conv(x))
+        return output
+
+
+class DownsampleLayer(nn.Module):
+    """
+    A downsampling layer that applies convolution, optional pooling, and activation.
+
+    Arguments
+    ---------
+    in_channels : int
+        Number of input channels.
+    out_channels : int
+        Number of output channels.
+    kernel_size : int
+        Size of the convolutional kernel.
+    stride : int, optional
+        Stride of the convolution (default is 1).
+    causal : bool, optional
+        If True, applies causal convolution (default is False).
+    activation : nn.Module, optional
+        Activation function (default is PReLU).
+    use_weight_norm : bool, optional
+        If True, applies weight normalization to the convolution (default is True).
+    pooling : bool, optional
+        If True, applies an average pooling operation (default is False).
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        stride: int = 1,
+        causal: bool = False,
+        activation=nn.PReLU(),
+        use_weight_norm: bool = True,
+        pooling: bool = False,
+    ):
+        super(DownsampleLayer, self).__init__()
+        self.pooling = pooling
+        self.stride = stride
+        self.activation = activation
+        self.use_weight_norm = use_weight_norm
+        if pooling:
+            self.layer = Conv1d(
+                in_channels, out_channels, kernel_size, causal=causal
+            )
+            self.pooling = nn.AvgPool1d(kernel_size=stride)
+        else:
+            self.layer = Conv1d(
+                in_channels,
+                out_channels,
+                kernel_size,
+                stride=stride,
+                causal=causal,
+            )
+        if use_weight_norm:
+            self.layer = weight_norm(self.layer)
+
+    def forward(self, x):
+        """
+        Applies convolution, optional pooling, and activation to the input data.
+
+        Arguments
+        ---------
+        x : torch.Tensor
+            Input tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            Processed output tensor.
+        """
+        x = self.layer(x)
+        x = self.activation(x) if self.activation is not None else x
+        if self.pooling:
+            x = self.pooling(x)
+        return x
+
+    def remove_weight_norm(self):
+        """
+        Removes weight normalization from the convolutional layer.
+        """
+        if self.use_weight_norm:
+            remove_weight_norm(self.layer)
+
+
+class UpsampleLayer(nn.Module):
```

Comment:
If I use the SB downsampling, it causes an error when loading the pretrained weights since the naming could be a bit different.  Since it is going to be moved to the integration folder, I suggest keep all their module as it is . 
I am working on another recepie that include many of these quantizati